src/node_crypto.cc

```diff
@@ -745,6 +747,73 @@ void SecureContext::SetKey(const FunctionCallbackInfo<Value>& args) {
   }
 }
 
+void SecureContext::SetEngineKey(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+  unsigned int len = args.Length();
+  if (len != 2) {
+    return THROW_ERR_MISSING_ARGS(env,
+                 "Private key name and engine name are required");
+  }
+
+  char errmsg[1024];
+  const node::Utf8Value engine_id(env->isolate(), args[1]);
+  ENGINE* e = LoadEngineById(*engine_id, &errmsg);
+  if (!e) {
+    return env->ThrowError(errmsg);
+  }
+
+  if (!ENGINE_init(e)) {
+    return env->ThrowError("ENGINE_init");
+  }
+
+  const node::Utf8Value key_name(env->isolate(), args[0]);
+  EVPKeyPointer key(ENGINE_load_private_key(e, *key_name, nullptr, nullptr));
+
+  if (!key) {
+    unsigned long err = ERR_get_error();  // NOLINT(runtime/int)
+    if (!err) {
+      return env->ThrowError("ENGINE_load_private_key");
+    }
+    return ThrowCryptoError(env, err);
+  }
+
+  int rv = SSL_CTX_use_PrivateKey(sc->ctx_.get(), key.get());
+
+  if (!rv) {
+    unsigned long err = ERR_get_error();  // NOLINT(runtime/int)
+    if (!err)
+      return env->ThrowError("SSL_CTX_use_PrivateKey");
+    return ThrowCryptoError(env, err);
+  }
+}
+
+void SecureContext::SetClientSigalgs(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  SecureContext* sc;
+  ASSIGN_OR_RETURN_UNWRAP(&sc, args.Holder());
+
+  unsigned int len = args.Length();
+  if (len != 1) {
+    return THROW_ERR_MISSING_ARGS(env,
+                "Colon-separated list of signature algoritms is required");
+  }
```

Comment:
Same as above: Use `CHECK_EQ(len, 1)` instead if this function is not supposed to be called directly.